src/plugins/vis_augmenter/public/saved_augment_vis/_saved_augment_vis.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @name SavedAugmentVis
+ *
+ * @extends SavedObject.
+ */
+import { get } from 'lodash';
+import {
+  createSavedObjectClass,
+  SavedObject,
+  SavedObjectOpenSearchDashboardsServices,
+} from '../../../saved_objects/public';
+import { IIndexPattern } from '../../../data/public';
+import { extractReferences, injectReferences } from './saved_augment_vis_references';
+
+export function createSavedAugmentVisClass(services: SavedObjectOpenSearchDashboardsServices) {
+  const SavedObjectClass = createSavedObjectClass(services);
+
+  class SavedAugmentVis extends SavedObjectClass {
+    public static type: string = 'augment-vis';
+    public static mapping: Record<string, string> = {
+      description: 'text',
+      pluginResourceId: 'text',
+      savedObjectType: 'keyword',
+      savedObjectId: 'keyword',
+      visLayerExpressionFn: 'object',
+      version: 'integer',
+    };
+
+    constructor(opts: Record<string, unknown> | string = {}) {
+      if (typeof opts !== 'object') {
+        opts = { id: opts };
+      }
+      super({
+        type: SavedAugmentVis.type,
+        mapping: SavedAugmentVis.mapping,
+        extractReferences,
+        injectReferences,
+        id: (opts.id as string) || '',
+        indexPattern: opts.indexPattern as IIndexPattern,
+        defaults: {
+          description: get(opts, 'description', ''),
+          pluginResourceId: get(opts, 'pluginResourceId', ''),
+          savedObjectType: get(opts, 'savedObjectType', ''),
+          savedObjectId: get(opts, 'savedObjectId', ''),
+          visLayerExpressionFn: get(opts, 'visLayerExpressionFn', {}),
+          version: 1,
+        },
+      });
+      // TODO: determine if this saved obj should be visible in saved_obj_management plugin.
+      // if not, we can set showInRecentlyAccessed to false and not persist any edit URL
+      // probably set to false since this saved obj should be hidden by default
```

Comment:
@joshuarrrr @ashwin-pc any opinions on making this saved object type visible in the saved objects management plugin or not? 

I'm leaning towards the latter, due to the fact that it's a bit more "internal" than existing saved object types, and isn't meant to be edited by users. Also it may scale to the point where there is a large amount of them that could clutter the UI.

Reply:
Yeah if we have a scaling issue, we can find UI solutions to mitigate that, but that page is an advanced feature, so limiting the users ability to manage saved objects by themselves feels limiting.

Reply:
Got it, thanks for the inputs. With that I agree it makes sense to expose it here; I can work on that as a follow-up in this PR.